src/libstd/num/f32.rs

```diff
@@ -1391,7 +1390,11 @@ mod tests {
         assert_eq!((-0f32).integer_decode(), (0, -150, -1));
         assert_eq!(INFINITY.integer_decode(), (8388608, 105, 1));
         assert_eq!(NEG_INFINITY.integer_decode(), (8388608, 105, -1));
-        assert_eq!(NAN.integer_decode(), (12582912, 105, 1));
+
+        // Ignore the "sign" (quiet / signalling flag) of NAN.
+        // It can vary between runtime operations and LLVM folding.
+        let (nan_m, nan_e, _nan_s) = NAN.integer_decode();
+        assert_eq!((nan_m, nan_e), (12582912, 105));
```

Comment:
I can’t believe this test worked for so long on all of the platforms we support, because mantissa of NaN is not specified to be anything in particular. The only requirement is that the MSB of the mantissa is set to 1 to differentiate from infinity.